app/tests/test_search_endpoint.py

```diff
@@ -139,14 +107,38 @@ def test_query_multiple_non_allowed_parameters(self):
         """
         response = self.client.post(self.path, data=payload, content_type="application/json")
         self.assertStatusCode(400, response)
-        json_data_post = response.json()
+        json_data = response.json()
         for wrong_par in [wrong_query_parameter1, wrong_query_parameter2]:
             self.assertIn(
                 wrong_par,
-                str(json_data_post['description']),
+                str(json_data['description']),
                 msg=f"Wrong query parameter {wrong_par} not found in error message"
             )
 
+    def test_limit_in_post(self):
+        # limit in payload
+        limit = 1
+        payload = {'query': {'created': {'lte': '9999-12-31T09:07:39.399892Z'}}, 'limit': limit}
+        response = self.client.post(self.path, data=payload, content_type="application/json")
+        self.assertStatusCode(200, response)
+        json_data_payload = response.json()
+        self.assertEqual(len(json_data_payload['features']), 1, msg="More than one item returned.")
+
+        # limit in url
+        payload.pop('limit')
+        response = self.client.post(
+            f"{self.path}?limit={limit}", data=payload, content_type="application/json"
+        )
+        self.assertStatusCode(200, response)
+        json_data_url = response.json()
+
+        # compare limit in payload and limit in url
+        self.assertEqual(
+            json_data_payload['features'][0]['id'],
+            json_data_url['features'][0]['id'],
+            msg="Limit parameter in payload and in URL yielded different responses."
+        )
```

Comment:
Here I would make a stronger comparison
```python
self.assertEqual(json_data_payload['features'], json_data_url['features'])
```
To make sure that both answer are exaclty the same, not only checking the first item id